slt/standard/functions/scalars/string/regexp_count.slt

```diff
@@ -0,0 +1,53 @@
+# regexp_count function
+
+query I
+SELECT regexp_count('abacad', 'a|b');
+----
+4
+
+query I
+SELECT regexp_count('abacad', 'a');
+----
+3
+
+query I
+SELECT regexp_count('abacad', 'z');
+----
+0
+
+query I
+SELECT regexp_count('', 'a');
+----
+0
+
+# Test with a complex regex pattern
+query I
+SELECT regexp_count('hello world hello', '\b\w+\b');
+----
+3
```

Comment:
Add a note here saying that postgres uses `\y` to match word boundaries. The equivalent query in postgres would be `SELECT regexp_count('hello world hello', '\y\w+\y');`

We don't support that since we're just using the rust regex crate.